### PR TITLE
Added Emoji to Miscellaneous

### PR DESCRIPTION
**Pull request by thomasfdm**
Added the ability to generate random Emoji.

These are Unicode characters from the block U+1F600 to U+1F637. 

Since many messages and text posts online now make use of Emoji (especially if send from a mobile device), I think some may find it useful.

### CODE REVIEW

**Comment by thomasfdm** [comment]
Didn't change any code from 2fd6fe9f19ab9fbb262c9a03f50948767b097824 .

Travis says the build failed because of part of the DataTime tests, should I make an issue for this?

File: src/Faker/Provider/Miscellaneous.php [hunk 1]
```diff
@@ -273,10 +300,21 @@ public static function languageCode()
...
     }
+
+    /**
+     * Returns an encoded Unicode Character between U+1F600 and U+1F637.
+     *
+     * @link https://en.wikipedia.org/wiki/Emoji#Unicode_blocks
+     */
+    public static function emoji()
+    {
+        return json_decode('"' . static::randomElement(static::$emoji) . '"');
```
Comment (fzaninotto) [inline src/Faker/Provider/Miscellaneous.php]: why use `json_decode` for that? Why not store unicode directly in the `static::$emoji` source? That would be faster.

Comment (thomasfdm) [inline src/Faker/Provider/Miscellaneous.php]: I did that in my original version 62b87b5de007df5dac40594df9258cbf397c4bbc , however the "\u{xxxx}" notation wasn't added to PHP until 7.0, so it failed the 5.x Travis builds.

After research this was apparently the best method to store unicode characters in PHP pre-7.0

**Comment by fzaninotto** [comment]
Then we'll merge it as is. Thank you!

**Pull request merged**